src/server/pachw/server/pachw.go

```diff
@@ -0,0 +1,117 @@
+package server
+
+import (
+	"context"
+	"path"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/backoff"
+	"github.com/pachyderm/pachyderm/v2/src/internal/dlock"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/middleware/auth"
+	"github.com/pachyderm/pachyderm/v2/src/internal/miscutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/task"
+	"github.com/pachyderm/pachyderm/v2/src/server/pfs/server"
+)
+
+const (
+	tickerSeconds      = 5
+	scaleDownThreshold = 60 / tickerSeconds
+	period             = time.Second * time.Duration(tickerSeconds)
+)
+
+type pachW struct {
+	log *logrus.Logger
+	env Env
+}
+
+func NewController(ctx context.Context, env *Env) {
+	controller := &pachW{
+		env: *env,
+		log: env.Logger,
+	}
+	go controller.run(ctx)
+}
+
+func (p *pachW) run(ctx context.Context) {
+	ctx = auth.AsInternalUser(ctx, "pachw-controller")
+	backoff.RetryUntilCancel(ctx, func() error { //nolint:errcheck
+		eg, ctx := errgroup.WithContext(ctx)
```

Comment:
Why is this errgroup needed?

Reply:
Also, it looks like we are not returning errors in a few places, just logging them. We should return errors within the backoff callback and lean on the logging in the backoff.

Reply:
I think we want an error group here so we can return the error from the goroutine. 

Not returning errors inside is a good point, I forgot that this was happening in a backoff.

Reply:
This is partially resolved: Errors are returned, but we still are using an error group.

Reply:
I don't think it needs to be in a goroutine, you can just run the code in the backoff callback. Also, we should capture the error during unlock with a multierror and the second scale call is missing a error return.

Reply:
That's fair, I guess right now the pachw controller doesn't need to process anything else, so it can just run this routine in the main thread.